server/container/guice/cassandra/src/main/java/org/apache/james/modules/MailboxFlagInconsistencyModule.java

```diff
@@ -0,0 +1,61 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules;
+
+import org.apache.james.mailbox.cassandra.mail.CassandraDeletedMessageDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMailboxRecentsDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMessageIdDAO;
+import org.apache.james.mailbox.cassandra.mail.task.SolveMailboxFlagInconsistenciesService;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+
+public class MailboxFlagInconsistencyModule extends AbstractModule {
+
+    interface MailboxFlagInconsistencyInjectionsKey {
+        String SOLVE_MAILBOX_RECENT = "SolveMailboxRecentFlag";
+        String SOLVE_MAILBOX_DELETED = "SolveMailboxDeletedFlag";
+    }
+
+    @Provides
+    @Singleton
+    @Named(MailboxFlagInconsistencyInjectionsKey.SOLVE_MAILBOX_RECENT)
+    public SolveMailboxFlagInconsistenciesService provideSolveMailboxRecentInconsistenciesService(CassandraMessageIdDAO messageIdDAO,
+                                                                                                  CassandraMailboxDAO mailboxDAO,
+                                                                                                  CassandraMailboxRecentsDAO mailboxRecentDAO) {
+        return new SolveMailboxFlagInconsistenciesService(
+            new SolveMailboxFlagInconsistenciesService.SoleMailboxRecentFlagInconsistenciesStrategy(mailboxRecentDAO),
+            messageIdDAO, mailboxDAO);
+    }
+
+    @Provides
+    @Singleton
+    @Named(MailboxFlagInconsistencyInjectionsKey.SOLVE_MAILBOX_DELETED)
+    public SolveMailboxFlagInconsistenciesService solveMailboxDeletedInconsistenciesService(CassandraMessageIdDAO messageIdDAO,
+                                                                                            CassandraMailboxDAO mailboxDAO,
+                                                                                            CassandraDeletedMessageDAO deletedMessageDAO) {
+        return new SolveMailboxFlagInconsistenciesService(
+            new SolveMailboxFlagInconsistenciesService.SolveMailboxDeletedFlagInconsistenciesStrategy(deletedMessageDAO),
+            messageIdDAO, mailboxDAO);
+    }
```

Comment:
As said I would have SolveMailboxFlagInconsistenciesService being @Injected with `SolveMailboxDeletedFlagInconsistenciesStrategy` and `SolveMailboxRecentFlagInconsistenciesStrategy` as parameters and choosing the one to be used based on the TargetFlag, specified as a method parameter on each call. 

That way this guice module is not needed.

Reply:
updated